src/lib/sys.c

```diff
@@ -93,12 +91,16 @@ int run_command_full_params(const char *stdout_file, const char *stderr_file, ch
 	}
 	close(null_fd);
 
-	if (replace_fd(STDOUT_FILENO, stdout_file) < 0) {
```

Comment:
We still need the replace_fd when stdout_file or stderr_file is set. We need them for:

    run_command_full("file1", "file2", "/usr/bin/ls", "/tmp", NULL);  
    run_command_quiet("/usr/bin/ls", "/tmp", NULL);

Reply:
Done